std/format.d

```diff
@@ -4423,6 +4423,9 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
     static if (__traits(hasMember, T, "toString") && isSomeFunction!(val.toString))
         static assert(!__traits(isDisabled, T.toString), T.stringof ~
             " cannot be formatted because its `toString` is marked with `@disable`");
+    static if (__traits(hasMember, T, "toString") && !hasToString!(T, Char))
```

Comment:
Will this identify the bug in the case where `T` has an additional, valid, `toString` implementation?  Example:

```D
struct S
{
    string toString()
    {
        // this is valid
    }

    void toString (Output) (ref Output output)
        if (isOutputRange!(Output, char))
    {
        // fails to instantiate because we didn't import `isOutputRange`
    }
}
```

In this case we'd still like the bug in the invalid second implementation to be flagged by the compiler.  Will it be?

Reply:
> Will this identify the bug in the case where T has an additional, valid, toString implementation?

Part of the reason I'm concerned about this is about predictability of behaviour: if one writes an output-range-based `toString` to allow formatting without generating a new `string` instance, but the `string toString()` is silently used under the hood instead, this may create hard-to-track-down problems for apps that care about avoiding garbage generation.

Reply:
It will not. We could special-case the existence of a non-templated overload with templated overloads that fail to instantiate, but that will complicate the code significantly, and I'm not sure there are no cases where that'd be the wrong thing to do.